testsuite/cluster-test/src/experiments/performance_benchmark.rs

```diff
@@ -103,15 +111,39 @@ impl Experiment for PerformanceBenchmark {
         instance::instancelist_to_set(&self.down_instances)
     }
 
-    async fn run(&mut self, context: &mut Context<'_>) -> anyhow::Result<()> {
+    async fn run(&mut self, context: &mut Context<'_>) -> Result<()> {
         let stop_effects: Vec<_> = self
             .down_instances
             .clone()
             .into_iter()
             .map(StopContainer::new)
             .collect();
-        let futures = stop_effects.iter().map(|e| e.activate());
-        join_all(futures).await;
+        if let Some(cluster_swarm) = context.cluster_swarm {
+            let instance_configs: Vec<&InstanceConfig> = self
+                .down_instances
+                .iter()
+                .map(|instance| -> Result<&InstanceConfig> {
```

Comment:
Since this is pretty complex pieces of duplicating code, can we make it a function in `instance.rs`?
Something like `instance_configs(&Vec<Instance>) -> Vec<& InstanceConfig>`